src/token/api/src/ledger.rs

```diff
@@ -41,7 +43,11 @@ impl Ledger {
             .history
             .iter()
             .rev()
-            .filter(|tx| who.map_or(true, |c| c == tx.from || c == tx.to || Some(c) == tx.caller))
+            .filter(|tx| {
+                who.map_or(true, |c| {
```

Comment:
add a `contains` method `TxRectord` and then update the filter to `filter(|&tx| who.map_or(true, |c| tx.contains(c)))`